src/rez/config.py

```diff
@@ -109,6 +109,10 @@ class OptionalStr(Str):
     schema = Or(None, basestring)
 
 
+class OptionalObject(Str):
+    schema = Or(None, object)
```

Comment:
This schema is a bit too loose; also the subclassing of `Str`, while it will work currently, is a bit shaky (Str._parse_env_var gives expected behaviour for OptionalObject, but that's not reason enough for the subclassing). It's also somewhat misleading for an `OptionalObject` to also support env-var parsing as a string, but not anything else. Suggest:

```
class OptionalStrOrFunction(Setting):
    schema = Or(None, str, callable)
    def _parse_env_var(self, value):
        # note: env-var override only supports string, eg 'mymodule.preprocess_func'
        return value
```